File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    n_one_count: int = format(n, 'b').count('1')
    num: int = n + 1
    while True:
        num_one_count: int = format(num, 'b').count('1')
        if n_one_count == num_one_count:
            return num
        num += 1
    
    return num

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        int num = n;
        int nOneCount = countOneInBinary(n);
        while (true) {
            num++;
            int numOneCount = countOneInBinary(num);
            if (nOneCount == numOneCount)
                break;
        }
        return num;
    }
    
    private int countOneInBinary(int num) {
        String numToBinary = Integer.toBinaryString(num);
        int numOneCount = 0;
        for (int i = 0; i < numToBinary.length(); i++) {
            if (numToBinary.charAt(i) == '1')
                numOneCount ++;
        }
        return numOneCount;
    }
}